nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -275,4 +303,13 @@ protected void writeBatch(String payload, FlowFile parent, ProcessContext contex
         session.getProvenanceReporter().receive(flowFile, getURI(context));
         session.transfer(flowFile, rel);
     }
+
+    protected void configureMapper(String setting) {
```

Comment:
I don't think this is thread-safe based on the PR for your JSON validator. We should move the mapper into onTrigger to be safe.